authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -0,0 +1,1006 @@
+                          matchPolicyItems.stream()
+                              .forEach(
+                                  policyItem -> {
+                                    // If the role is not in the policy item, then add it
+                                    if (!policyItem.getRoles().contains(roleName)) {
+                                      policyItem.getRoles().add(roleName);
+                                    }
+                                  });
+                        }
+                      });
+            });
+  }
+
+  /**
+   * Remove policy item base the securable object's privileges and role name. <br>
+   * We didn't directly clean the policy items, because one Ranger policy maybe contain multiple
+   * Gravitino privilege objects. <br>
+   */
+  private void removePolicyItem(
+      RangerPolicy policy, String roleName, SecurableObject securableObject) {
+    // First check the privilege if support in the Ranger Hive
+    checkSecurableObject(securableObject);
+
+    // Delete the policy role base the securable object's privileges
+    policy.getPolicyItems().stream()
+        .forEach(
+            policyItem -> {
+              policyItem
+                  .getAccesses()
+                  .forEach(
+                      access -> {
+                        boolean matchPrivilege =
+                            securableObject.privileges().stream()
+                                .filter(Objects::nonNull)
+                                .flatMap(privilege -> translatePrivilege(privilege.name()).stream())
+                                .filter(Objects::nonNull)
+                                .anyMatch(
+                                    privilege -> {
+                                      return access.getType().equals(privilege);
+                                    });
+                        if (matchPrivilege
+                            && !policyItem.getUsers().isEmpty()
+                            && !policyItem.getGroups().isEmpty()) {
+                          // Not ownership policy item, then remove the role
+                          policyItem.getRoles().removeIf(roleName::equals);
+                        }
+                      });
+            });
+
+    // Delete the policy items if the roles is empty and not ownership policy item
+    policy
+        .getPolicyItems()
+        .removeIf(
+            policyItem -> {
+              return policyItem.getRoles().isEmpty()
+                  && policyItem.getUsers().isEmpty()
+                  && policyItem.getGroups().isEmpty();
+            });
+  }
+
+  private boolean checkSecurableObject(SecurableObject securableObject) {
+    securableObject
+        .privileges()
+        .forEach(
+            privilege -> {
+              check(
+                  checkPrivilege(privilege.name()),
+                  "This privilege %s is not support in the Ranger hive authorization",
```

Comment:
Some unnecessary privileges will be pushed down underlying systems.

Reply:
What unnecessary privileges? please specify

Reply:
`UseSchema` `CreateCatalog` will be pushed to Ranger, too.

Reply:
@xunliu Could you reply this?